### PR TITLE
Companion PR for ` Add a `build-sync-spec` subcommand and remove the CHT roots from the light sync state.`

### PR DESCRIPTION
**Pull request by expenses**
See https://github.com/paritytech/substrate/pull/6999. Not strictly a companion PR, as the substrate PR doesn't break anything. It would still be nice to get them merged at the same time though.

### CODE REVIEW

**Comment by expenses** [comment]
@cecton  do you think you could quickly review this please? Thanks!

File: cli/src/command.rs [hunk 1]
```diff
@@ -154,6 +155,39 @@ pub fn run() -> Result<()> {
...
+			runner.async_run(|config| {
+				let chain_spec = config.chain_spec.cloned_box();
+				let network_config = config.network.clone();
+				let service::NewFull { task_manager, client, network_status_sinks, .. }
+					= service::new_full_nongeneric(
+						config, None, authority_discovery_enabled, grandpa_pause, false,
+					)?;
+				let client = Arc::new(client);
+
+				Ok((cmd.run(chain_spec, network_config, client, network_status_sinks), task_manager))
```
Comment (seunlanlege) [inline cli/src/command.rs:188]: why not just pass the `Configuration` struct? `chain_spec` and `network_config` live on the `Configuration` object

Comment (expenses) [reply cli/src/command.rs:188]: The `Configuration` gets consumed by `new_full_nongeneric`, so it can't be used.

Comment (seunlanlege) [reply cli/src/command.rs:188]: I would argue that it should be clone, but that's for another time. This looks solid to me

**Review by seunlanlege — APPROVED** [review]
looks solid

**Review by cecton — APPROVED** [review]
Seems good to me

**Comment by expenses** [comment]
bot merge

**Comment by expenses** [comment]
bot merge

**Comment by expenses** [comment]
bot merge

**Pull request merged**